mir/src/optimize.rs

```diff
@@ -10,10 +10,11 @@ pub fn optimize(q: MirQuery) -> MirQuery {
     q
 }
 
-pub fn optimize_post_reuse(q: &mut MirQuery) {
-    find_and_merge_filter_chains(q);
+pub fn optimize_post_reuse(_q: &mut MirQuery) {
+    // find_and_merge_filter_chains(q);
```

Comment:
Why did we disable this optimization? I guess it affects reuse in security policies?

Reply:
Ah, I remember it was merging some filters that it shouldn't have and I meant to look into it, but ended up forgetting about it.